src/motorError.cpp

```diff
@@ -0,0 +1,84 @@
+#include "motorError.h"
+
+Timeout evaluationTimeout;
+Timeout resetTimeout;
+
+InterruptIn motorErrorPin(D0);
+
+#define SEQUENCE_DELAY 1000ms
+#define RESET_DELAY 3000ms
+
+// The error type, determined based on input pulses
+ErrorType errorType = None;
+
+// The pulse count, incremented with every rising edge of the motor error pin
+int pulses = 0;
+
+// Resets error type to None if no pulses have been recorded for the last RESET_DELAY
+void reset()
+{
+    if (pulses == 0)
+    {
+        errorType = None;
+    }
+}
+
+// Evaluates the motor error type based on the current number of pulses, resets pulse count
+void evaluatePulses()
+{
+    switch (pulses)
+    {
+        case 0: errorType = None; break;
+        case 1: errorType = Error1; break;
+        case 3: errorType = Error3; break;
+        case 4: errorType = Error4; break;
+        case 5: errorType = Error5; break;
+        case 6: errorType = Error6; break;
+        case 8: errorType = Error8; break;
+        case 9: errorType = Error9; break;
+        default: errorType = Unknown; break;
+    }
+    pulses = 0;
+
+    // reset() will be called after delay RESET_DELAY
+    resetTimeout.attach(&reset, RESET_DELAY);
+
+    // Note: when evaluatePulses() is called again before a prior delay has elapsed, the
+    // prior delay is disregarded; reset() will only be called at the end of the new delay
+}
+
+// Increments the pulse count and calls the evaluation method with delay SEQUENCE_DELAY
+void newPulse()
+{
+    pulses ++;
+
+    // evaluatePulses() will be called after delay SEQUENCE_DELAY
+    evaluationTimeout.attach(&evaluatePulses, SEQUENCE_DELAY);
+
+    // Note: when newPulse() is called again before a prior delay has elapsed, the prior
+    // delay is disregarded; evaluatePulses() will only be called at the end of the new delay
+}
+
+// Start motor error interpretation
+void startErrorInterpretation()
+{
+    // newPulse() is called with every rising edge of the motor error pin
+    motorErrorPin.rise(&newPulse);
+}
+
+// Represents the error type as a string
+const char * errorString(ErrorType errorType)
+{
+    switch (errorType)
+    {
+        case None: return "None";
+        case Error1: return "Error 1";
+        case Error3: return "Error 3";
+        case Error4: return "Error 4";
+        case Error5: return "Error 5";
+        case Error6: return "Error 6";
+        case Error8: return "Error 8";
+        case Error9: return "Error 9";
+        case Unknown: return "Unknown";
```

Comment:
Same as in header, would be nice if these are a bit more descriptive with the actual error names from the [spreadsheet](https://docs.google.com/spreadsheets/d/12O2UPdM_fqUVKd0IXZ638wb1obtBXQp11giv7swfWM4/edit#gid=1209182212)